### PR TITLE
fix lint issues & add type-checking for Sanity studio projects #1637

### PR DESCRIPTION
**Pull request by millianapia**

### CODE REVIEW

File: sanityv3/src/previews/PreviewWrapper.tsx [hunk 1]
```diff
@@ -54,3 +54,4 @@ export const PreviewWrapper = ({ src, children, shareable = true }: Props) => {
     </StyledWrapper>
   )
 }
+export default PreviewWrapper
```
Comment (fernandolucchesi) [inline sanityv3/src/previews/PreviewWrapper.tsx:57]: why was this needed? 🤔

File: sanityv3/src/lib/structure/index.js [hunk 2]
```diff
@@ -55,45 +55,28 @@ const SUB_EDITOR_ITEMS = (S) =>
...
-  ].filter((e) => e)
+
   if (isAdmin) {
-    return ADMIN_ITEMS(S)
+    return ADMIN_ITEMS(S, context)
   } else if (isSubEditor) {
-    return SUB_EDITOR_ITEMS(S)
+    return SUB_EDITOR_ITEMS(S, context)
   } else if (isLocalNewsEditor) {
     return LOCAL_NEWS_EDITOR_ITEMS(S)
```
[the remark this answers is not in the document]
Comment (millianapia) [reply sanityv3/src/lib/structure/index.js:81]: is that necessary when LOCAL_NEWS_EDITOR_ITEMS does not use context? 

`const LOCAL_NEWS_EDITOR_ITEMS = (S) => [LocalNews(S)].filter((e) => e)
`

File: sanityv3/src/previews/PreviewWrapper.tsx (hunk 1 above)
Comment (millianapia) [reply sanityv3/src/previews/PreviewWrapper.tsx]: to be honest, i do not know, reversed this

File: sanityv3/schemas/components/SlugInput/utils/useAsync.ts [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import React from 'react'
+import {DependencyList, useState, useRef, useCallback} from 'react'
```
[the remark this answers is not in the document]
Comment (millianapia) [reply sanityv3/schemas/components/SlugInput/utils/useAsync.ts]: space has now been added and the file formatted

File: sanityv3/plugins/asset-source-brandmaster/src/BrandmasterAssetSource.tsx [hunk 4]
```diff
@@ -19,6 +19,7 @@ const BrandmasterAssetSource = forwardRef<HTMLDivElement>((props: any, ref) => {
 
   const handleBrandmasterEvent = useCallback(
     (event) => {
+      console.log('brandmaster', event)
```
[the remark this answers is not in the document]
Comment (millianapia) [reply sanityv3/plugins/asset-source-brandmaster/src/BrandmasterAssetSource.tsx]: it is now purged

File: sanityv3/src/lib/structure/index.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (fernandolucchesi) [reply sanityv3/src/lib/structure/index.js:81]: yes, it is wrong because it should use context, since LocalNews use context:

`const LocalNews = (S, context) => Flags.HAS_LOCAL_NEWS && LocalNewsStructure(S, context)`

**Review by fernandolucchesi — CHANGES_REQUESTED** [review]
requires changes to line 56... we should rly look into enabling eslint for v3

**Review by fernandolucchesi — APPROVED** [review]

**Comment by fernandolucchesi** [comment]
looks good however i need to look into why ide is not highlighting issues later
